pennylane/transforms/optimization/undo_swaps.py

```diff
@@ -109,8 +109,6 @@ def qfunc():
 
     gates.reverse()
 
-    new_tape = type(tape)(
-        gates, tape.measurements, shots=tape.shots, trainable_params=tape.trainable_params
-    )
+    new_tape = tape.copy(operations=gates)
```

Comment:
Was copying the `trainable_params` a mistake here? Changing the operations will set the `trainable_params` back to default, whereas here it looks like we were preserving them explicitly.